### MODULE-7: PYTHON SESSION1 TASK

SESSION 1 - Introduction to Python & InstallationTopics:What is Python? Why it’s essential for analyticsInstalling Python, Anaconda, Jupyter NotebookUnderstanding .py vs .ipynbWriting your first Python cellDemo:Print “Hello Analyst” in Jupyter Notebook

1. Install Anaconda on your system, open Anaconda Navigator, and launch Jupyter Notebook from it.

2. Create a new Jupyter Notebook file named analytics_intro.ipynb and write a cell that prints your name and your favorite app (like Instagram or Zomato) using the print() function.

In [2]:
print("uma")

uma


In [3]:
print('google')

google


3. Open VS Code or any code editor, create a Python script file called hello_app.py, and write code to print 'Welcome to Data Analytics with Python!'. Then, run this script from the terminal.

In [4]:
print("Welcome to Data Analytics with Python!")

Welcome to Data Analytics with Python!


4. In Jupyter Notebook, add a new cell and write Python code to print the current year using the datetime module.<br><br><em><strong>Hint:</strong> Import datetime and use datetime.datetime.now().year.</em>

In [5]:
import datetime

print(datetime.datetime.now().year)

print(datetime.datetime.now())

2026
2026-06-25 21:33:06.211172


### MODULE-7: PYTHON SESSION2 TASK

SESSION 2 - Python Syntax, Keywords, Variables & CommentsTopics:Indentation rulesNaming conventionsComments (# and multiline)Variables: dynamic typingDemo:Create variables for sales, region, discount

1. Create three variables in Python: user_name, fav_app, and daily_usage_hours. Assign your own name, your favorite app (like Instagram or Zomato), and how many hours you use it daily.

In [6]:
user_name = 'UMA'
fav_app = 'GOOGLE'
daily_usage_hours = 24

print(user_name)
print(fav_app)
print(daily_usage_hours)

UMA
GOOGLE
24


2. Write a Python script that declares variables for product_name, price, and is_available to represent an item on Flipkart. Print each variable and its data type using the type() function.

In [7]:
product_name = 'ac_remote'
price = 400
is_available = True

print(product_name, price, is_available)
print(type(product_name))
print(type(price))
print(type(is_available))

ac_remote 400 True
<class 'str'>
<class 'int'>
<class 'bool'>


3. Demonstrate the difference between single-line and multi-line comments in Python by writing a script that explains how a Spotify playlist recommendation system might work. Use # for single-line and triple quotes for multi-line comments.

In [8]:
# Spotify tracks the songs a user listens to.

# It identifies favorite genres and artists.

# Based on listening patterns, it recommends similar songs.

"""
Spotify's recommendation system uses user behavior,
such as liked songs, skipped tracks, and playlist activity.
It compares this information with data from other users
to suggest new songs and personalized playlists.
"""

print("Spotify Playlist Recommendation System")

Spotify Playlist Recommendation System


4. Create variables for order_total, delivery_region, and discount_percent to represent a Zomato order. Follow Python naming conventions and print a sentence using all three variables, like 'Order from [region] totals ₹[order_total] with [discount_percent]% discount.'

In [9]:
order_total = 2
delivery_region = 'abc'
discount_percent = 10

print(f"Order from {delivery_region} totals ₹{order_total} with {discount_percent}% discount.")

Order from abc totals ₹2 with 10% discount.


5. Write a Python script that intentionally mixes tabs and spaces for indentation, then fix the script so it runs without errors.<br><br><em><strong>Hint:</strong> Use only spaces for indentation, as per Python's best practices.</em>

In [10]:
# incorrect program:
for i in range(3):
	print(i)        # indented with tab
    print("Hello")  # indented with spaces

IndentationError: unindent does not match any outer indentation level (<string>, line 4)

In [ ]:
# correct program 
for i in range(3):
    print(i)
    print("hello")
    

0
hello
1
hello
2
hello


### MODULE-7: PYTHON SESSION3 TASK

SESSION 3 - Data Types in PythonTopics:int, float, str, booltype() functionType castingDemo:Convert numeric strings to floats for analysis

1. Declare four variables in Python: one integer (number of followers), one float (average rating), one string (your favorite app's name), and one boolean (is_premium_user). Print each variable and its type using the type() function.

In [ ]:
followers = 5555
average_rating = 4.4
fav_app = "youtube"
is_premium_user = False

print(followers)
print(type(followers))

print(average_rating)
print(type(average_rating))

print(fav_app)
print(type(fav_app))

print(is_premium_user)
print(type(is_premium_user))


5555
<class 'int'>
4.4
<class 'float'>
youtube
<class 'str'>
False
<class 'bool'>


2. Write a Python program that takes a user's input for the price of a Zomato order as a string, converts it to a float using type casting, adds 18% GST, and prints the final bill amount.

In [ ]:
price = input("Enter the bill amount")
gst_price = float(int(price) + (int(price)*0.18))
print(gst_price)

1180.0


In [ ]:
m = ("34")
type(m)

str

In [ ]:
l=float(m)
l

34.0

In [ ]:
type(l)

float

3. Given a list of strings representing product prices from Flipkart, like ['199.99', '299.50', '150'], convert all to floats and calculate the total cart value.

In [ ]:
l = ['199.99', '299.50', '150']
l2= []

In [ ]:
for i in l:
    j = float(i)
    l2.append(j)

In [ ]:
l2

[199.99, 299.5, 150.0]

In [ ]:
sum = 0
for i in l2:
    sum = sum+i
print(sum)

649.49


In [ ]:
from functools import reduce
a = reduce (lambda a, b: a+b, l2)
a

649.49

4. Build a function is_discount_applicable(order_amount) that takes a float and returns True if the amount is greater than 500, otherwise False. Print the result for order amounts 450 and 750.

In [ ]:
def discount(amt):
    if amt > 500:
        return True
    else:
        return False

In [ ]:
discount(450)

False

In [ ]:
discount(750)

True

5. You received a dataset of ratings as strings from Spotify: ['4.5', '3.0', '5', '4.2']. Use type casting to convert these to floats, then find and print the highest rating.<br><br><em><strong>Hint:</strong> Use the float() function inside a loop or list comprehension.</em>

In [ ]:
rating = ['4.5', '3.0', '5', '4.2']

rating2 = []

for i in rating:
    j = float(i)
    rating2.append(j)

print(rating2)
highest_rating = max(rating2)
print("highest_rating : ", highest_rating)

[4.5, 3.0, 5.0, 4.2]
highest_rating :  5.0


### MODULE-7: PYTHON SESSION4 TASK

SESSION 4 - StringsTopics:String methods: upper, lower, replace, splitIndexing & slicingFormatting stringsDemo:Clean product names like " samsung-Mobile "

1. Create a Python script that takes any product name string (e.g., 'Redmi Note 12 Pro') and prints the name in all uppercase and all lowercase using the upper() and lower() methods.

In [ ]:
product_name = 'Redmi Note 12 Pro'

capital = product_name.upper()
smaller = product_name.lower()
print(capital)
print(smaller)

REDMI NOTE 12 PRO
redmi note 12 pro


2. Write a function clean_brand_name(name) that removes leading/trailing spaces and replaces any hyphens '-' with a single space in the input string. Test it with ' oneplus-Nord '.

In [ ]:
def clean_brand_name(name):
    cleaned = name.strip(name)          # .strip() → removes spaces from the beginning and end.
    cleaned = name.replace('-', ' ')

    return cleaned

In [ ]:
result = clean_brand_name(' oneplus-Nord ')
print(result)

 oneplus Nord 


3. Given the string 'Apple iPhone 14 Pro Max', use string slicing to extract and print only the brand name and the model (i.e., 'Apple' and 'iPhone 14 Pro Max') separately.<br><br><em><strong>Hint:</strong> Use split() to help find the split point, then use slicing for the substrings.</em>

In [ ]:
str = "Apple-iPhone 14 PRO Max"

abc = str[0:5] #slicing
print(abc)

xyz = "Apple-iPhone 14 PRO Max".split('-') #split
print(xyz)

pqr = str[6:]
print(pqr)

Apple
['Apple', 'iPhone 14 PRO Max']
iPhone 14 PRO Max


4. Build a function format_product_display(name, price) that takes a product name and price (e.g., 'Boat Earbuds', 1299) and returns a formatted string like 'Boat Earbuds - ₹1299'.

In [ ]:
def format_product_display(name, price):
    format = f'{name} - ₹{price}'
    return format

In [ ]:
format_product_display('Boat Earbuds', 1299)

'Boat Earbuds - ₹1299'

In [ ]:
formatted = format_product_display('Boat Earbuds', 1299)
print(formatted)

Boat Earbuds - ₹1299


5. Suppose you have a list of messy product names: [' mi-Band 5 ', ' SAMSUNG-Galaxy ', ' realme-Book ']. Write code to clean each name (remove spaces, replace hyphens with spaces, and make the brand title case) and print the cleaned list.<br><br><em><strong>Constraint:</strong> Use at least three string methods from this session.</em>

In [16]:
product_names = [' mi-Band 5 ', ' SAMSUNG-Galaxy ', ' realme-Book ']

cleaned_product_names = []

for name in product_names:
    clean_name = name.strip()
    clean_name = clean_name.replace("-", " ")
    clean_name = clean_name.title()

    cleaned_product_names.append(clean_name)

print(cleaned_product_names)

['Mi Band 5', 'Samsung Galaxy', 'Realme Book']


In [17]:
product_names = [' mi-Band 5 ', ' SAMSUNG-Galaxy ', ' realme-Book ']
names = []
replaced = []
updated_product_names = []

for i in product_names:
    text = i
    #print(text.strip())
    names.append(text.strip())

for i in names:
    name = i
    replaced.append(name.replace('-',' '))

for i in replaced:
    str = i
    updated_product_names.append(str.title())
    
print(names)
print(replaced)
print(updated_product_names)

['mi-Band 5', 'SAMSUNG-Galaxy', 'realme-Book']
['mi Band 5', 'SAMSUNG Galaxy', 'realme Book']
['Mi Band 5', 'Samsung Galaxy', 'Realme Book']


### MODULE-7: PYTHON SESSION5 TASK

SESSION 5 - Lists & TuplesTopics:List operations: append, extend, popTuple immutabilityWhen to use lists vs tuplesDemo:Create a list of customer IDs, update and iterate

1. Create a list called playlist_ids with 5 integers representing Spotify playlist IDs, then use append() to add a new playlist ID at the end and print the updated list.

In [3]:
playlist_ids = [1, 2, 3, 4, 5]
new_id = 6

playlist_ids.append(new_id)
print(playlist_ids)

[1, 2, 3, 4, 5, 6]


In [ ]:
id = [8, 7, 9]

playlist_ids.append(id)
print(playlist_ids)
# This is adding the whole list at the end

[1, 2, 3, 4, 5, 6, [8, 7, 9]]


`append()` - List - Adds one item at the end

`extend()` - List - Adds multiple items from another iterable

`update()` - Dictionary, Set - Updates existing items or adds new ones

In [ ]:
# using for loop i added/appended the elements of one list to another list
# Extend is used to add elements of a list to another list
lst = [101, 102, 103, 104, 105]

lst2 = [201, 202, 203, 204]
'''
for i in lst2:
    lst.append(i)

print(lst)
'''
lst2.extend(lst)
print(lst2)

[201, 202, 203, 204, 101, 102, 103, 104, 105]


2. Simulate a Flipkart shopping cart: start with a list cart_items containing 't-shirt', 'shoes'. Use extend() to add ['jeans', 'cap'] to the cart, then print the final list of items.

In [7]:
cart_items = ['t-shirt', 'shoes']
a = ['jeans', 'cap']

cart_items.extend(a)
print(cart_items)

['t-shirt', 'shoes', 'jeans', 'cap']


3. Write a function remove_last_item(order_list) that pops the last item from a Zomato order list and returns the removed item. Test it with a sample order_list.

In [19]:
def remove_last_item(order_list):
    a = order_list.pop()
    print(order_list)
    return a

lst4 = [101, 102, 103, 104, 105]
b = remove_last_item(lst4)
print(b)

[101, 102, 103, 104]
105


4. Create a tuple called insta_filters with 4 Instagram filter names. Try to update the second filter and observe what error you get. Explain in a comment why this happens.<br><br><em><strong>Hint:</strong> Tuples are immutable, so direct assignment won't work.</em>

In [2]:
insta_filters = ("filter1", "filter2", "filter3", "filter4")
l = list(insta_filters)
l

['filter1', 'filter2', 'filter3', 'filter4']

In [3]:
l[2] = "filter5"

In [4]:
l

['filter1', 'filter2', 'filter5', 'filter4']

In [6]:
t = tuple(l)
t

('filter1', 'filter2', 'filter5', 'filter4')

In [7]:
#another method
fruits = ("apple", "Mango", "berries", "guava")

In [8]:
frts = fruits[:2] + ("muskmelon",) + fruits[3:]

In [9]:
frts

('apple', 'Mango', 'muskmelon', 'guava')

5. Given two scenarios — storing a user's favorite genres (which may change) and storing a fixed set of IRCTC train classes ('Sleeper', 'AC 3 Tier', 'AC 2 Tier') — choose whether to use a list or tuple for each. Write one sentence explaining your choice for both.

In order to store user's fav genres which can be mutable, List should be taken as lists are mutable.

on the opposite side, for the fixed train classes which are immutable, Tuple should be selected as tuples are not flexible.

### MODULE-7: PYTHON SESSION6 TASK

SESSION 6 - Dictionaries & SetsTopics:Key-value pairsAdding, updating, deletingSet operations (union, intersection)Demo:Create product ? price dictionary

1. Create a Python dictionary called playlist_prices with at least 5 key-value pairs where the key is a Spotify playlist name (as a string) and the value is the playlist's price (as an integer). Print the dictionary.

In [10]:
playlist_prices = {
    "k1":50,
    "k2":60,
    "k3":70,
    "k4":20
    }

In [11]:
playlist_prices

{'k1': 50, 'k2': 60, 'k3': 70, 'k4': 20}

2. Write a function update_playlist_price(playlist, new_price) that updates the price of a given playlist in the playlist_prices dictionary. Test it by updating the price of any one playlist and printing the updated dictionary.

In [ ]:
def update_plylist_prc(playlist, new_price):

3. Remove a playlist from the playlist_prices dictionary using the del statement. Print the dictionary after deletion to confirm the change.

4. Given two sets: set1 contains the names of restaurants you have ordered from on Zomato, and set2 contains the names of restaurants you have ordered from on Swiggy, find and print the union and intersection of these sets.<br><br><em><strong>Hint:</strong> Use the union() and intersection() methods of Python sets.</em>

### MODULE-7: PYTHON SESSION7 TASK

SESSION 7 - Conditional StatementsTopics:if, elif, elseNested conditionsDemo:If revenue > target ? print “Achieved”

1. Write a Python script that checks if a user's entered age is 18 or above and prints 'Eligible for IPL ticket booking' if true, otherwise prints 'Not eligible'.

2. Create a Python program that takes the number of followers as input and uses if, elif, and else to print 'Micro Influencer' if followers < 10,000, 'Rising Star' if between 10,000 and 100,000, and 'Celebrity' if above 100,000.

3. Build a Python script that asks the user for their Zomato order total and prints 'Apply Free Delivery' if total is above 299, 'Add more items for free delivery' if between 200 and 299, else 'Delivery charges apply'.

4. Write a Python program using nested if statements: take a user's entered Flipkart cart value and payment method ('UPI', 'Card', 'Cash'). If the cart value is above 1000 and payment method is 'UPI', print 'Eligible for 10% cashback'; if above 1000 and payment is not 'UPI', print 'Eligible for 5% cashback'; else print 'No cashback'.

### MODULE-7: PYTHON SESSION8 TASK

SESSION 8 - LoopsTopics:for loopwhile loopbreak / continueEnumerateDemo:Loop through sales list ? calculate total

1. Create a Python list called order_amounts with the values [120, 250, 90, 310, 150]. Use a for loop to calculate and print the total order value.

2. Given a list of cricket scores [45, 78, 102, 34, 67, 89], use a while loop to print each score until you reach a score above 100, then stop printing.

3. Simulate a Flipkart cart with a list of item prices [299, 499, 199, 999, 149]. Use a for loop and the continue statement to skip any item priced below 200, and print the total of the remaining items.

4. You have a list of favorite song names: ['Kesariya', 'Believer', 'Shape of You', 'Blinding Lights', 'Excuses']. Use the enumerate() function in a for loop to print each song with its playlist position (starting from 1).

5. Write a for loop that goes through a list of Instagram follower counts [120, 1500, 23000, 800, 45000] and prints 'Micro', 'Influencer', or 'Celebrity' for each, based on the following: Micro (<1000), Influencer (1000-10000), Celebrity (>10000).<br><br><em><strong>Hint:</strong> Use if-elif-else inside the loop to check the follower count range.</em>